In [1]:
%matplotlib inline
import matplotlib.pyplot as plt
from pycocotools.coco import COCO
from pycocotools.cocoeval import COCOeval
import numpy as np
import skimage.io as io
import pylab
pylab.rcParams['figure.figsize'] = (10.0, 8.0)

In [2]:
annType = ['segm','bbox','keypoints']
annType = annType[1]      #specify type here
prefix = 'person_keypoints' if annType=='keypoints' else 'instances'
print ('Running demo for *%s* results.'%(annType))

Running demo for *bbox* results.


# 데이터셋 경로 여기에 넣어야함
---

In [3]:
#initialize COCO ground truth api
visionNoob_dataDir='J:/git_projects/V_brain/examples/toy_dataset'
heefe_dataDir = 'put your path'

dataDir = visionNoob_dataDir

---

In [4]:
dataType='train2018'
#annFile = '%s/annotations/%s_%s.json'%(dataDir,prefix,dataType)
annFile = '%s/%s_KNU_%s.json'%(dataDir,prefix,dataType)
resFile = '%s/%s_KNU_%s_results.json'%(dataDir,prefix,dataType)

print(annFile)
print(resFile)

cocoGt=COCO(annFile)
cocoDt=cocoGt.loadRes(resFile)

J:/git_projects/V_brain/examples/toy_dataset/instances_KNU_train2018.json
J:/git_projects/V_brain/examples/toy_dataset/instances_KNU_train2018_results.json
loading annotations into memory...
Done (t=0.00s)
creating index...
index created!
Loading and preparing results...
DONE (t=0.00s)
creating index...
index created!


In [7]:
imgIds = sorted(cocoGt.getImgIds())
imgId = imgIds[np.random.randint(5)]
catIds = cocoGt.getCatIds()

In [8]:
# running evaluation
cocoEval = COCOeval(cocoGt,cocoDt,'bbox')
cocoEval.params.imgIds  = imgIds#이거 조절하면 특정 데이터 알아낼 수 있다. 
cocoEval.params.catIds = catIds #이거 조절하면 특정 아이디 알아낼 수 있다. 
cocoEval.evaluate()
cocoEval.accumulate()
cocoEval.summarize()

Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=0.00s).
Accumulating evaluation results...
DONE (t=0.01s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.950
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 1.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.950
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.950
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.950
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.950
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=

 # 결과 분석
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.950  
 AP가 0.95인 이유는 내가 클래스 1 어떤애 스코어를 조금 바꿔놔서 그럼  
 클래스 1을 빼놓고 구하면 1이 나와야함

In [11]:
catIds = cocoGt.getCatIds()
catIds = catIds[1:3]
print(catIds)

[2, 3]


In [12]:
# running evaluation
cocoEval = COCOeval(cocoGt,cocoDt,'bbox')
cocoEval.params.imgIds  = imgIds#이거 조절하면 특정 데이터 알아낼 수 있다. 
cocoEval.params.catIds = catIds #이거 조절하면 특정 아이디 알아낼 수 있다. 
cocoEval.evaluate()
cocoEval.accumulate()
cocoEval.summarize()

Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=0.00s).
Accumulating evaluation results...
DONE (t=0.00s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 1.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 1.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=